framework/src/actions/BatchMeshGeneratorAction.C

```diff
@@ -0,0 +1,554 @@
+      convertAndSetNumericScalar<unsigned short>(params, param_name, param_value);
+      break;
+    case (ParameterType::INT):
+      convertAndSetNumericScalar<int>(params, param_name, param_value);
+      break;
+    case (ParameterType::UINT):
+      convertAndSetNumericScalar<unsigned int>(params, param_name, param_value);
+      break;
+    case (ParameterType::ENUM):
+      if (params.isType<MooseEnum>(param_name))
+        params.set<MooseEnum>(param_name) = param_value;
+      break;
+    case (ParameterType::STRING):
+      if (params.isType<std::string>(param_name))
+        params.set<std::string>(param_name) = param_value;
+      break;
+    case (ParameterType::BOOL):
+      if (params.isType<bool>(param_name))
+        hit::toBool(param_value, &params.set<bool>(param_name));
+      break;
+    default:
+      mooseError("impossible situation."); // as we use MultiMooseEnum to ensure the type is valid
+  }
+}
+
+void
+BatchMeshGeneratorAction::setVectorParams(InputParameters & params,
+                                          const std::string & param_name,
+                                          const ParameterType & param_type,
+                                          const std::vector<std::string> & param_value)
+{
+  switch (param_type)
+  {
+    case (ParameterType::REAL):
+      convertAndSetNumericVector<Real>(params, param_name, param_value);
+      break;
+    case (ParameterType::SHORT):
+      convertAndSetNumericVector<short>(params, param_name, param_value);
+      break;
+    case (ParameterType::USHORT):
+      convertAndSetNumericVector<unsigned short>(params, param_name, param_value);
+      break;
+    case (ParameterType::INT):
+      convertAndSetNumericVector<int>(params, param_name, param_value);
+      break;
+    case (ParameterType::UINT):
+      convertAndSetNumericVector<unsigned int>(params, param_name, param_value);
+      break;
+    case (ParameterType::ENUM):
+      if (params.isType<std::vector<MooseEnum>>(param_name))
+        params.set<MultiMooseEnum>(param_name) = param_value;
+      break;
+    case (ParameterType::STRING):
+      if (params.isType<std::vector<std::string>>(param_name))
+        params.set<std::vector<std::string>>(param_name) = param_value;
+      break;
+    case (ParameterType::BOOL):
+      if (params.isType<std::vector<bool>>(param_name))
+      {
+        std::vector<bool> values(param_value.size());
+        std::transform(param_value.begin(),
+                       param_value.end(),
+                       values.begin(),
+                       [](const std::string & val)
+                       {
+                         bool tmp;
+                         hit::toBool(val, &tmp);
+                         return tmp;
+                       });
+        params.set<std::vector<bool>>(param_name) = values;
+      }
+      break;
+    default:
+      mooseError("impossible situation."); // as we use MultiMooseEnum to ensure the type is valid
```

Comment:
```suggestion
      mooseAssert(false, "impossible situation."); // as we use MultiMooseEnum to ensure the type is valid
```
so you dont lose a line of opt coverage